README_cmake.md

```diff
@@ -304,17 +695,219 @@ and the `USE_RUNTIME` option is ignored. Other options work as expected.
+| `CMAKE_BUILD_TYPE`              | Only applies to single-config generators.     | `$<CONFIG>`                                                                       |
+| `CMAKE_*_FLAGS*` (w/o `_INIT`)  | User-only                                     | Write a toolchain file with the corresponding `_INIT` variable                    |
+| `CMAKE_COLOR_MAKEFILE`          | User-only                                     | None                                                                              |
+| `CMAKE_ERROR_DEPRECATED`        | User-only                                     | None                                                                              |
+| `CMAKE_CONFIGURATION_TYPES`     | We only support the four standard build types | None                                                                              |
+
+Of course feel free to insert debugging helpers _while developing_ but please
+remove them before review. Finally, the following variables are allowed, but
+their use must be motivated:
+
+| Variable               | Reason                                              | Alternative                                                            |
+| ---------------------- | --------------------------------------------------- | ---------------------------------------------------------------------- |
+| `CMAKE_SOURCE_DIR`     | Points to global source root, not Halide's.         | `Halide_SOURCE_DIR` or `PROJECT_SOURCE_DIR`                            |
+| `CMAKE_BINARY_DIR`     | Points to global build root, not Halide's           | `Halide_BINARY_DIR` or `PROJECT_BINARY_DIR`                            |
+| `CMAKE_MAKE_PROGRAM`   | CMake abstracts over differences in the build tool. | Prefer CTest's build and test mode or CMake's `--build` mode           |
+| `CMAKE_CROSSCOMPILING` | Often misleading.                                   | Inspect relevant variables directly, eg. `CMAKE_SYSTEM_NAME`           |
+| `BUILD_SHARED_LIBS`    | Could override user setting                         | None, but be careful to restore value when overriding for a dependency |
+
+Any use of these functions and variables will block a PR.
+
+## Adding tests
+
+When adding a file to any of the folders under `test`, be aware that CI expects
+that every `.c` and `.cpp` appears in the `CMakeLists.txt` file _on its own
+line_, possibly as a comment. This is to avoid globbing and also to ensure that
+added files are not missed.
+
+For most test types, it should be as simple as adding to the existing lists,
+which must remain in alphabetical order. Generator tests are trickier, but
+following the existing examples is a safe way to go.
+
+## Adding apps
+
+If you're contributing a new app to Halide: great! Thank you! There are a few
+guidelines you should follow when writing a new app.
+
+- Write the app as if it were a top-level project. You should call
+  `find_package(Halide)` and set the C++ version to 14.
```

Comment:
Why set C++ to 14? (Earlier in this doc we show setting the C++ version to 11-or-later so this is a bit contradictory)

Reply:
Sorry, there is some lingering confusion about which C++ version to use. All of our supported compilers support C++14, so it is safe to request this either way. I will go through and switch things to C++11 until we officially flip the switch to 14.

Reply:
I really want us to bypass 14 and go straight to 17. Preferably soon.